packages/select/src/select.vue

```diff
@@ -42,9 +42,39 @@
           disable-transitions>
           <span class="el-select__tags-text">{{ item.currentLabel }}</span>
         </el-tag>
+        <input
+          key="input"
+          type="text"
+          class="el-select__input"
+          :class="[selectSize ? `is-${ selectSize }` : '']"
+          :disabled="selectDisabled"
+          :autocomplete="autoComplete || autocomplete"
+          :aria-controls="id ? `${id}-listbox` : null"
+          role="combobox"
+          aria-haspopup="listbox"
+          :aria-owns="id ? `${id}-listbox` : null"
+          :aria-expanded="visible ? 'true' : 'false'"	
+          @focus="handleFocus"
+          @blur="softFocus = false"
+          @keyup="managePlaceholder"
+          @keydown="resetInputState"
+          @keydown.down.prevent="navigateOptions('next')"
+          @keydown.up.prevent="navigateOptions('prev')"
+          @keydown.enter.prevent="selectOption"
+          @keydown.esc.stop.prevent="visible = false"
+          @keydown.delete="deletePrevTag"
+          @keydown.tab="handleTabKey"
+          @compositionstart="handleComposition"
+          @compositionupdate="handleComposition"
+          @compositionend="handleComposition"
+          v-model="query"
+          @input="debouncedQueryChange"
+          v-if="filterable"
+          :style="{ 'flex-grow': '1', width: selected.length > 0 ? (query.length > 0 ? query.length + 2 + 'ch':'2ch' ) : '100%', 'max-width': '100%' }"
```

Comment:
Move the `width` calculation into a computed since it is a bit hairy for inlining into the template.

Reply:
Moved into computed, multiSelectInputWidth